felix/collector/goldmane/client.go

```diff
@@ -56,18 +65,85 @@ func (g *GoldmaneReporter) Start() error {
 	g.once.Do(func() {
 		// We don't wait for the initial connection to start so we don't block the caller.
 		g.client.Connect(context.Background())
+
+		if g.mayReportToNodeSocket {
+			go g.nodeSocketReporter()
+		}
 	})
 	return err
 }
 
+func (g *GoldmaneReporter) nodeSocketReporter() {
+	for {
+		if NodeSocketExists() {
+			g.mayStartNodeSocketReporter()
+		} else {
+			g.mayStopNodeSocketReporter()
+		}
+		time.Sleep(time.Second * 10)
```

Comment:
I think it would be cleaner to use a feixconfig option for this instead of using a timer.